tests/test_elcometer456_block.py

```diff
@@ -32,14 +32,19 @@ def test_default_read(self):
             self.configure_block(blk, {})
             blk.start()
             # wait for notify_signals
-            e.wait(1.5)
+            e.wait(0.1)
         blk.stop()
-        self.assertDictEqual(
-            self.last_notified[DEFAULT_TERMINAL][0].to_dict(), {
-                'value': self.value
-            }
-        )
-        blk._serial.write.assert_called_with(b"O")
+        if self.value:
+            # signals are only notified if data value is valid
+            self.assertDictEqual(
+                self.last_notified[DEFAULT_TERMINAL][0].to_dict(), {
+                    'value': self.value
+                }
+            )
+            blk._serial.write.assert_called_with(b"O")
+        else:
+            # write only happens if data has been read
+            blk._serial.write.assert_not_called()
```

Comment:
This relates to my above comment. I think we do still want to do this on this reading `b'     ---      F1    \r\n'`